examples/functions_and_traits.rs

```diff
@@ -0,0 +1,178 @@
+//! Examples of how to write functions and traits that operate on `ndarray` types.
```

Comment:
This runs only when we run specifically ask for it with `cargo run --example ...`? Shouldn't it always run when we run the tests? Or it's already safe because the rest of the codebase uses everything here and more?

Reply:
I think it's fine - I originally had it in tests, but I realized it was more helpful to act as an example. Our CI/CD still runs the examples, so it'll be "checked" in that sense even if it's not a test.